packages/compass-crud/src/components/document.jsx

```diff
@@ -1,40 +1,41 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
+import HadronDocument from 'hadron-document';
 import EditableDocument from './editable-document';
 import ReadonlyDocument from './readonly-document';
 
-/**
- * Component for a single document in a list of documents.
- */
-class Document extends React.Component {
-  /**
-   * Render a single document list item.
-   *
-   * @returns {React.Component} The component.
-   */
-  render() {
-    if (this.props.editable && this.props.isTimeSeries) {
-      return (
-        <ReadonlyDocument
-          doc={this.props.doc}
-          copyToClipboard={this.props.copyToClipboard}
-          openInsertDocumentDialog={this.props.openInsertDocumentDialog}
-        />
-      );
-    }
-    if (this.props.editable) {
-      return (<EditableDocument {...this.props} />);
+const Document = (props) => {
+  const {
+    editable,
+    isTimeSeries,
+    copyToClipboard,
+    openInsertDocumentDialog,
+    doc: _doc,
+  } = props;
+
+  const doc = useMemo(() => {
+    if (_doc.type === 'Document') {
```

Comment:
Does this mean that the `doc` prop can already be a HadronDocument instance? When does that happen? Inside of hadron-document I've kind of tried to align the brand checking for the `Document` class to be a `.isRoot()` call, would it make sense to use that here as well?

Reply:
Ah, actually yes, that would be much better, thanks for the reminder this exists, I'll update.

In CRUD view plugin we do keep an actual instance of HadronDocument in redux store and do not create it in render so the instance is always the same and only updates when you edit the document so I was trying to kinda allow component interface to handle both use-cases: where we pass an instance that we manually manage outside or where we just have a "random" bson document that we want to pretty render somewhere in the app. I am not 100% happy with the solution, but also couldn't come up with a better approach that would require less changes tbh